### PR TITLE
Bug 1172013 - [Calendar] views should not depend directly on day observer r=gaye

### PR DESCRIPTION
**Pull request by millermedeiros**
just so we can move most of the day observer logic into the worker.

I'll probably provide another pull request with a temporary implementation of the day observer inside the backend folder (keeping all the DB actions but skipping the "controllers" logic)

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=db7700f861ed632d4c462c3378211ea32b647ade

File: apps/calendar/test/unit/views/single_day_test.js [hunk 1]
```diff
@@ -68,11 +65,11 @@ suite('Views.SingleDay', function() {
     subject.onactive();
     subject.onactive();
```
Comment (gaye) [inline apps/calendar/test/unit/views/single_day_test.js]: eh?

File: apps/calendar/test/unit/views/single_day_test.js [hunk 2]
```diff
@@ -81,60 +78,69 @@ suite('Views.SingleDay', function() {
...
-  });
+    setup(function() {
+      subject._observer = observer = { cancel: sinon.spy() };
+    });
 
-  test('#append', function() {
-    subject.setup();
-    subject.append();
+    test('> when inactive', function() {
+      subject.oninactive();
```
Comment (gaye) [inline apps/calendar/test/unit/views/single_day_test.js]: Eh?

File: apps/calendar/test/unit/views/single_day_test.js [hunk 3]
```diff
@@ -81,60 +78,69 @@ suite('Views.SingleDay', function() {
...
-      daysHolder.innerHTML,
-      `<div class="md__day" data-date="${d}"></div>`,
-      'should add day node'
-    );
+  suite('#append', function() {
+    setup(function(done) {
+      subject.setup();
+      subject.append();
+      // render is async!
+      setTimeout(done);
```
Comment (gaye) [inline apps/calendar/test/unit/views/single_day_test.js]: Maybe for clarity `nextTick` would be better?

File: apps/calendar/test/unit/views/single_day_test.js [hunk 4]
```diff
@@ -157,290 +163,302 @@ suite('Views.SingleDay', function() {
...
+      // busytime #1
+      var busy = busytimes[0];
+
+      // position/size is enough to test overlaps
+      assert.equal(busy.style.height, '99.9px', '1: height');
+      assert.equal(busy.style.top, '200px', '1: top');
+      assert.equal(busy.style.width, '50%', '1: width');
+      assert.equal(busy.style.left, '50%', '1: left');
+      assert.include(
+        busy.href, '/event/show/' + data.basic[0].busytime._id, '1: href');
```
Comment (gaye) [inline apps/calendar/test/unit/views/single_day_test.js]: Prefer template strings!

File: apps/calendar/test/unit/views/single_day_test.js [hunk 5]
```diff
@@ -157,290 +163,302 @@ suite('Views.SingleDay', function() {
...
+      assert.include(labels, makeFirstEventID('location'), '1: label location');
+      assert.include(labels, makeFirstEventID('icon'), '1: label icon');
+      assert.include(labels, makeFirstEventID('description'), '1: label desc');
+
+      assert.include(busy.className, 'has-alarms', '1: class');
+      assert.include(busy.className, 'has-overlaps', '1: class');
+
+      var remote = data.basic[0].event.remote;
+
+      var title = busy.querySelector('#' + makeFirstEventID('title'));
```
Comment (gaye) [inline apps/calendar/test/unit/views/single_day_test.js]: Same thing here and elsewhere about using template strings

File: apps/calendar/test/unit/views/single_day_test.js [hunk 6]
```diff
@@ -157,290 +163,302 @@ suite('Views.SingleDay', function() {
...
+      assert.lengthOf(busytimes, 2, '5: events');
+      assert.lengthOf(
+        alldaysHolder.querySelectorAll('.md__event'), 0, '5: all'
+      );
+      assert.equal(
+        busytimes[1].querySelector('.md__event-title').textContent.trim(),
+        data.basic[1].event.remote.title,
+        '5: event title'
+      );
+    });
```
Comment (gaye) [inline apps/calendar/test/unit/views/single_day_test.js]: This is a 250 line test. Is there any way that we can break this up? I think the ideal test is a one line assertion

File: apps/calendar/test/unit/views/single_day_test.js (hunk 1 above)
Comment (millermedeiros) [inline apps/calendar/test/unit/views/single_day_test.js]: to make sure that calling it twice in a row only triggers it once..

File: apps/calendar/test/unit/views/single_day_test.js (hunk 2 above)
Comment (millermedeiros) [inline apps/calendar/test/unit/views/single_day_test.js]: same thing about making sure that 2 consecutive calls only trigger it once.

File: apps/calendar/test/unit/views/single_day_test.js (hunk 3 above)
Comment (millermedeiros) [inline apps/calendar/test/unit/views/single_day_test.js]: shouldn't make any difference since we have the comment.

File: apps/calendar/test/unit/views/single_day_test.js (hunk 4 above)
Comment (millermedeiros) [inline apps/calendar/test/unit/views/single_day_test.js]: why? it's a single `+`

File: apps/calendar/test/unit/views/single_day_test.js (hunk 5 above)
Comment (millermedeiros) [inline apps/calendar/test/unit/views/single_day_test.js]: `+ foo` uses less chars than `${foo}`.. I didn't change anything on these tests, just the indentation (since I grouped into a `suite`)

File: apps/calendar/test/unit/views/single_day_test.js (hunk 6 above)
Comment (millermedeiros) [inline apps/calendar/test/unit/views/single_day_test.js]: I did not change the tests, just the indentation (grouped into a `suite` because of `setup/teardown`).

test is 250LOC because it is testing a lot of things at once.. we couldn't do the same thing with a single assertion without comparing the full string (that was the old logic and we always had to update the tests at each change to the markup - including whitespaces).

TL;DR; tests are good enough as is. trying to break it into 30+ tests makes no sense in this case. (we are making sure that the markup updates every time the data changes), as separate tests it would require way more setup/teardown logic.
